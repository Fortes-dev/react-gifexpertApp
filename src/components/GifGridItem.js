import React from 'react';
import PropTypes from 'prop-types';

export const GifGridItem = ( {title, url} ) => {

    return (

        //En vez de class, se usa className en react
        <div className="card animate__animated animate__backInLeft">
            <img src={url} alt={title}/>
            <p>{title}</p>
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}
