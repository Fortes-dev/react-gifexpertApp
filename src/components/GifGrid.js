import React, { useState, useEffect } from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';

export const GifGrid = ({category}) => {

    const {data, loading} = useFetchGifs(category); 

    return (
        <> 
        <h3 className="animate__animated animate__pulse">{category}</h3>
        { loading && <p className="animate__animated animate__fadeOut">Loading...</p>}
        <div className="card-grid">
                {
                    data.map( img => 
                        (
                            <GifGridItem 
                                {...img}
                                key={img.id}/>
                        )
                    )
                }
           
        </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}
