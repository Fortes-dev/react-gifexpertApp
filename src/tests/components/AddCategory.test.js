import { shallow } from "enzyme";
import '@testing-library/jest-dom';
import { AddCategory } from "../../components/AddCategory";


describe('Test suite de AddCategory', () => { 
    

    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={setCategories}/>);

    beforeEach ( () => {

        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories}/>);
    });

    test('Debe mostrar el snapshop del componente', () => { 
        expect(wrapper).toMatchSnapshot();

    });

    test('Debe de cambiar la caja de texto', () => { 
        
        const input = wrapper.find('input');
        const value = 'Hola mundo';

        // Simulamos el evento que es un objeto. Dentro del evento tenemos el target
        // Un atributo de ese objeto, y ya en el target le pasamos el value (el texto)
        input.simulate('change', { target: {value} });

     });

    test('NO debe de postear la informacion con submit', () => { 
         
        wrapper.find('form').simulate('submit', { preventDefault: () => {} });

        expect( setCategories ).not.toHaveBeenCalled();

    });

    test('Debe de llamar al setCategories y limpiar la caja de texto', () => { 
        
        const value = 'unodostres';
        const input = wrapper.find('input')

        input.simulate('change', { target: {value} });
        
        wrapper.find('form').simulate('submit', { preventDefault: () => {} });

        expect( setCategories ).toHaveBeenCalled();
        expect( setCategories ).toHaveBeenCalledWith( expect.any(Function) );
        expect( input.prop('value') ).toMatch('');

    });

});
