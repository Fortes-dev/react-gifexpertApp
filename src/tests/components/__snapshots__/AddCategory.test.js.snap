// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite de AddCategory Debe mostrar el snapshop del componente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
